fix(ci): release workflow for preview releases and pypi publish

BeArchiTek opened the pull request:
This PR adds pypi publishing of infrahub-server during a release.

It also fixes issues we had with the current release workflow: it did not take account of all syntax cases for the version, especially when releasing a preview release. This PR switches the workflow to leverage the Github release event, check if it's a github pre-release, validates if the (pre)release matches the python version.

It also fixes an edge case where the "latest" docker tag would be assigned to older versions, we now make use of the "latest" github release flag.

BeArchiTek committed:
add pypi publish

ogenstad commented on .github/workflows/publish-pypi.yml:
```diff
@@ -0,0 +1,67 @@
+        uses: "actions/cache@v4"
+        id: "cached-poetry-dependencies"
+        with:
+          path: "./infrahub_sync/.venv"
```
I'm guessing this is copied from another repo.

BeArchiTek replied:
spotted on Patrick :p 
It was a kickstart for @fatih-acar  to continue but yeah bad copy-paste

fatih-acar committed:
fix(ci): update release workflow to use release events

This fixes issues with pre-releases where the git tag may not match the
previous filter we had.

We now check if the github release is a "pre-release" and also check if
the Poetry version is also a pre-release or dev-release one.

If the github release is not pre-release, we ensure that the Poetry
version is also not a pre-release one.

We also fix a potential issue where the "latest" Docker tag would be
added to older versions. Leverage the GitHub "latest" release flag to
solve that.

Signed-off-by: Fatih Acar <fatih@opsmill.com>

fatih-acar committed:
fix(ci): ignore some linter rules

Signed-off-by: Fatih Acar <fatih@opsmill.com>

fatih-acar commented:
Edited the PR description with new changes around the release workflow.
cc @lykinsbd

BeArchiTek reviewed (commented):
LGTM
